### PR TITLE
chore: add pool link queries

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

Adds:

```
osmosisd q gamm cl-pool-link-from-cfmm [cfmm-pool-id]
osmosisd q concentratedliquidity cfmm-pool-link-from-cl [cl-pool-id]
```

### CODE REVIEW

File: x/gamm/client/cli/query.go [hunk 1]
```diff
@@ -325,6 +326,16 @@ Example:
 	)
 }
 
+// GetConcentratedPoolIdLinkFromCFMMRequest returns concentrated pool id that is linked to the given cfmm pool id.
+func GetConcentratedPoolIdLinkFromCFMMRequest() (*osmocli.QueryDescriptor, *types.QueryConcentratedPoolIdLinkFromCFMMRequest) {
+	return &osmocli.QueryDescriptor{
+		Use:   "cl-pool-link-from-cfmm-pool-id [poolID]",
```
Comment (AlpinYukseloglu) [inline x/gamm/client/cli/query.go:332]: nit: this feels a little verbose – thoughts on removing the `-pool-id` suffix from both of these queries?

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, just had a minor non-blocking comment

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/gamm/client/cli/query.go (hunk 1 above)
Comment (p0mvn) [reply x/gamm/client/cli/query.go:332]: Agreed. I will apply the change to speed things up

File: x/concentrated-liquidity/client/cli/query.go [hunk 2]
```diff
@@ -82,3 +83,12 @@ func GetIncentiveRecords() (*osmocli.QueryDescriptor, *queryproto.IncentiveRecor
 {{.CommandPrefix}} incentive-records 1`,
 	}, &queryproto.IncentiveRecordsRequest{}
 }
+
+func GetCFMMPoolIdLinkFromConcentratedPoolId() (*osmocli.QueryDescriptor, *queryproto.CFMMPoolIdLinkFromConcentratedPoolIdRequest) {
+	return &osmocli.QueryDescriptor{
+		Use:   "cfmm-pool-link-from-cl-pool-id [poolId]",
```
Comment (p0mvn) [inline x/concentrated-liquidity/client/cli/query.go:89]:
```suggestion
		Use:   "cfmm-pool-link-from-cl [poolId]",
```

File: x/gamm/client/cli/query.go (hunk 1 above)
Comment (p0mvn) [reply x/gamm/client/cli/query.go:332]: Done, please merge when CI is passing

File: proto/osmosis/concentrated-liquidity/query.proto [hunk 3]
```diff
@@ -98,6 +98,16 @@ service Query {
...
         "/osmosis/concentratedliquidity/v1beta1/tick_accum_trackers";
   };
+
+  // CFMMPoolIdLinkFromConcentratedPoolId returns the pool id of the CFMM
+  // pool that is linked with the given concentrated pool.
+  rpc CFMMPoolIdLinkFromConcentratedPoolId(
+      CFMMPoolIdLinkFromConcentratedPoolIdRequest)
+      returns (CFMMPoolIdLinkFromConcentratedPoolIdResponse) {
+    option (google.api.http).get =
+        "/osmosis/gamm/v1beta1/cfmm_pool_id_link_from_concentrated_pool_id/"
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply proto/osmosis/concentrated-liquidity/query.proto]: Good catch, addressed here https://github.com/osmosis-labs/osmosis/pull/5465/commits/f84d33b30ad4f2096b917786a3533cc7abc1d7ab

**Pull request merged**